homeassistant/components/application_credentials/__init__.py

```diff
@@ -234,6 +238,27 @@ async def _async_provide_implementation(
     ]
 
 
+async def async_post_remove_config_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+) -> dict[str, Any] | None:
+    """Return the item id of an application credential for an existing ConfigEntry."""
+    if not await _get_platform(hass, config_entry.domain) or not (
+        auth_domain := config_entry.data.get("auth_implementation")
+    ):
+        return None
+
+    storage_collection = hass.data[DOMAIN][DATA_STORAGE]
+    for item in storage_collection.async_items():
+        item_id = item[CONF_ID]
+        if (
+            item[CONF_DOMAIN] == config_entry.domain
+            and item.get(CONF_AUTH_DOMAIN, item_id) == auth_domain
+        ):
+            return {"application_credential_id": item_id}
```

Comment:
I'd use the domain as prefix.
```suggestion
            return {"application_credentials_id": item_id}
```

Reply:
The item id has a domain as the prefix fwiw.

Are you suggesting something like this?
`{config_entry.domain: { "application_credentials_id": item_id}}`
?

Reply:
No, I just meant application_credentials vs application_credential.